pkg/kubectl/cmd/cmd.go

```diff
@@ -158,7 +158,9 @@ __custom_func() {
 
 	// If you add a resource to this list, please also take a look at pkg/kubectl/kubectl.go
 	// and add a short forms entry in expandResourceShortcut() when appropriate.
+	// TODO: This should be populated using the discovery information from apiserver.
 	valid_resources = `Valid resource types include:
+   * clusters (valid only when talking to federation apiserver)
```

Comment:
(valid only for federation apiservers)

reads a bit better.  "talking" is awkward in some contexts.